MapboxNavigation/EndOfRouteViewController.swift

```diff
@@ -0,0 +1,161 @@
+import UIKit
+import MapboxDirections
+
+fileprivate enum ConstraintSpacing: CGFloat {
+    case closer = 8.0
+    case further = 65.0
+}
+
+class EndOfRouteViewController: UIViewController {
+
+    //MARK: - IBOutlets
+    @IBOutlet weak var labelContainer: UIView!
+    @IBOutlet weak var primary: UILabel!
+    @IBOutlet weak var endNavigationButton: UIButton!
+    @IBOutlet weak var stars: RatingControl!
+    @IBOutlet weak var commentView: UITextView!
+    @IBOutlet weak var commentViewContainer: UIView!
+    @IBOutlet weak var showCommentView: NSLayoutConstraint!
+    @IBOutlet weak var hideCommentView: NSLayoutConstraint!
+    @IBOutlet weak var ratingCommentsSpacing: NSLayoutConstraint!
+    
+    //MARK: - Properties
+    lazy var placeholder: String = NSLocalizedString("Add an optional comment here.", comment: "Comment Placeholder Text")
+    lazy var endNavigation: String = NSLocalizedString("End Navigation", comment: "End Navigation Button Text")
+    
+    var dismiss: ((Int, String?) -> Void)?
+    var comment: String?
+    var rating: Int = 0 {
+        didSet {
+            rating == 0 ? hideComments() : showComments()
+        }
+    }
+    
+    open var destination: Waypoint? {
+        didSet {
+            guard isViewLoaded else { return }
+            updateInterface()
+        }
+    }
+
+    //MARK: - Lifecycle Methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        clearInterface()
+        stars.didChangeRating = { (new) in self.rating = new }
+        setPlaceholderText()
+        styleCommentView()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        roundCornersOfRootView()
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+    }
+
+    //MARK: - IBActions
+    @IBAction func endNavigationPressed(_ sender: Any) {
+        dismissView()
+    }
+    
+    //MARK: - Private Functions
+    private func roundCornersOfRootView() {
```

Comment:
Can we move this to the UIView extension? Perhaps merge it with `applyDefaultCornerRadiusShadow`?

Reply:
- [ ] Extract to extension